samples/meetings-transcription/csharp/MeetingTranscription/Bots/TranscriptionBot.cs

```diff
@@ -0,0 +1,147 @@
+        /// Stores the Azure configuration values.
+        /// </summary>
+        private readonly IOptions<AzureSettings> azureSettings;
+
+        /// <summary>
+        /// Store details of meeting transcript.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, string> transcriptsDictionary;
+
+        /// <summary>
+        /// Instance of card factory to create adaptive cards.
+        /// </summary>
+        private readonly ICardFactory cardFactory;
+
+        /// <summary>
+        /// Creates bot instance.
+        /// </summary>
+        /// <param name="azureSettings">Stores the Azure configuration values.</param>
+        /// <param name="transcriptsDictionary">Store details of meeting transcript.</param>
+        /// <param name="cardFactory">Instance of card factory to create adaptive cards.</param>
+        public TranscriptionBot(IOptions<AzureSettings> azureSettings, ConcurrentDictionary<string, string> transcriptsDictionary, ICardFactory cardFactory)
+        {
+            this.transcriptsDictionary = transcriptsDictionary;
+            this.azureSettings = azureSettings;
+            graphHelper = new GraphHelper(azureSettings);
+            this.cardFactory = cardFactory;
+        }
+
+        /// <summary>
+        /// Activity handler for on message activity.
+        /// </summary>
+        /// <param name="turnContext">A strongly-typed context object for this turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var replyText = $"Echo: {turnContext.Activity.Text}";
+            await turnContext.SendActivityAsync(MessageFactory.Text(replyText, replyText), cancellationToken);
+        }
+
+        /// <summary>
+        /// Activity handler for meeting end event.
+        /// </summary>
+        /// <param name="meeting">The details of the meeting.</param>
+        /// <param name="turnContext">A strongly-typed context object for this turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected override async Task OnTeamsMeetingEndAsync(MeetingEndEventDetails meeting, ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var meetingInfo = await TeamsInfo.GetMeetingInfoAsync(turnContext);
+
+            var result = await graphHelper.GetMeetingTranscriptionsAsync(meetingInfo.Details.MsGraphResourceId);
+            if (result != string.Empty)
```

Comment:
NullOrEmpty check here too?

Reply:
As I am returning only string.Empty it is not mandatory to check for null. But as best practice I have updated the code.